cli/src/git_util.rs

```diff
@@ -155,16 +156,69 @@ pub fn with_remote_git_callbacks<T>(
     callbacks.progress = progress_callback
         .as_mut()
         .map(|x| x as &mut dyn FnMut(&git::Progress));
+    // Based on Git's implementation: https://github.com/git/git/blob/43072b4ca132437f21975ac6acc6b72dc22fd398/sideband.c#L178
     let mut sideband_progress_callback = None;
+    let mut scratch = Vec::new();
+    let display_prefix = "remote: ";
+    let suffix = if std::io::stdout().is_terminal() {
+        "\x1B[K"
+    } else {
+        "        "
+    };
     if print_sideband_messages {
-        sideband_progress_callback = Some(|progress: &[u8]| {
-            let message = String::from_utf8_lossy(progress);
-            if message.contains("\n") {
-                for line in message.lines() {
-                    _ = writeln!(ui.lock().unwrap().stderr(), "remote: {line}");
+        sideband_progress_callback = Some(|progress_message: &[u8]| {
+            let mut index = 0;
+            // Append a suffix to each nonempty line to clear the end of the screen line.
+            loop {
+                let Some(i) = progress_message[index..]
+                    .find_char('\n')
+                    .or_else(|| progress_message[index..].find_char('\r'))
+                    .map(|i| index + i)
```

Comment:
Did you mean `i + 1` here?

Reply:
It's supposed to be `index + i` here, because I'm looking for the next instance of the character after `index`, but there didn't seem to be a method in the stdlib which did that. I used `find_char` which was in gitoxide's `ByteSlice` trait, but also had to slice from the `index` offset onwards, so to get the location of the character in the original slice I did `index + i`.

Do let me know if there's a better way to do this.